File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
    int n = nums.size();
    
    int maxi=0;
    unordered_set<int>st(nums.begin(),nums.end());
    for(int i:st){
        int cnt=1;
        if(st.find(i-1)==st.end()){
            int x=i;
            while(st.find(x+1)!=st.end()){
                cnt++;
                x++;
               
            }
        }
        maxi=max(maxi,cnt);
    }   
    return maxi;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        n= len(nums)
        maxi=-1
        st=set(nums) 
        for i in st:
            if i-1 not in st:
                cnt=0
                x=i
                while x+1 in st:
                    cnt=cnt+1
                    x=x+1
                maxi=max(maxi,cnt)

        return maxi+1